Add switch playbook test in the CI

jmaupetit opened the pull request:
## Purpose

As mentioned in #190, the `switch` playbook is a crucial part of our blue-green deployment strategy. We need to automate its testing to avoid breaking it during our developments.

## Proposal

- [x] Add a `ci-switch` script
- [x] Integrate this script for each tested app in CircleCI's configuration

Fixes #190

jmaupetit committed:
🎨(ci) test service prefix now defaults to current

Make the ci-test-service script more explicit by defaulting the prefix
to current and explicitly point to the next route when required. This
seems less convoluted.

Also, I've removed the "set -x" debug flag that is not intented in
production but for development.

lunika commented on bin/ci-test-service:
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 #
-# usage: ci-test-service [SERVICE] [CONTENT] [PATH]
-set -x
```
haha sorry for this one :rofl:

jmaupetit replied:
:ok_hand:

jmaupetit committed:
✨(ci) add tests for the switch playbook

Routes switch is the core part of our blue-green deployment strategy and
its testing was not automated. This work improves the situation by
systematically testing it for every distrubuted blue-green app.

Fix #190

lunika commented on .circleci/config.yml:
```diff
@@ -169,7 +169,17 @@ jobs:
+          command: |
+            # Switch next route to current
+            bin/ci-switch "hello"
+            # Test service switched to the next route
```
the current no ?

lunika commented on .circleci/config.yml:
```diff
@@ -215,7 +225,17 @@ jobs:
+          command: |
+            # Switch next route to current
+            bin/ci-switch "richie"
+            # Test service switched to the next route
```
s/next/current/

lunika commented on .circleci/config.yml:
```diff
@@ -238,7 +258,17 @@ jobs:
+          command: |
+            # Switch next route to current
+            bin/ci-switch "forum"
+            # Test service switched to the next route
```
s/next/current/

lunika commented on .circleci/config.yml:
```diff
@@ -261,7 +291,18 @@ jobs:
+          command: |
+            # Switch next route to current
+            bin/ci-switch "edxapp"
+            # Test service switched to the next route
```
s/next/current/

lunika commented on .circleci/config.yml:
```diff
@@ -285,7 +326,17 @@ jobs:
+          command: |
+            # Switch next route to current
+            bin/ci-switch "marsha"
+            # Test service switched to the next route
```
s/next/current/

lunika commented on bin/_ci_config.sh:
```diff
@@ -0,0 +1,27 @@
+# This utility improves the developer experience by dynamically setting
+# environment variables required to run ansible playbooks and play with
+# an OpenShift instance from Arnold's container (see Dockerfile).
+function _docker_run() {
```
why not reusing the function in `_config.sh` file ? This function is configurable, you can change the `env_file` used and the `TTY` option.

jmaupetit replied:
Yes, but we also mount sources as a volume, I wanted to avoid making the original function too complicated.

jmaupetit replied:
I've made the `docker_run` command more complex to use as you asked :trollface:

lunika reviewed (approved):
![](https://media.giphy.com/media/BIuuwHRNKs15C/giphy.gif)
